emsdk

```diff
@@ -2053,13 +2053,53 @@ def process_tool_list(tools_to_activate, log_errors=True):
   return tools_to_activate
 
 
+def run_emcc(tools_to_activate):
```

Comment:
It feels slightly hacky to me to be relying on running emcc just to get this side effect. Especially given that we are explicitly determining the cache directory below, so I don't really see the benefit of doing this over just manually removing the cache directory.

Reply:
Removing the cache directory isn't enough, unless I don't understand what you're proposing.

The emsdk *does* remove the cache dir. The problem is that the first time emcc runs it will remove it *again* - because it sees the config file changed. To avoid that, we run it the first time here. Only then can we safely add contents to the cache.

Is the comment text not clear enough?

Reply:
Oh, I see. The comment below implies to me that we are running emcc in order to clear the cache, but really we are running it to avoid having it clear the cache on the *next* run. How does emcc decide that the cache needs to be cleared? just the modified time on some file?

Reply:
Yes, the settings file. If it sees it changed, it clears the cache.

I'll improve the comment.